crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -1538,3 +1541,259 @@ async fn create_val_fullnode_executor(
+        let test_account = TestAccount {
+            address,
+            key_pair,
+            gas: obj.id(),
+        };
+        objects.push(obj);
+        test_account
+    };
+
+    // For each named Sui account without an address value, create an account with an adddress
+    // and a gas object
+    for n in account_names {
+        let test_account = mk_account();
+        account_objects.insert(n.clone(), test_account.gas);
+        accounts.insert(n, test_account);
+    }
+
+    // Make a default account with a gas object
+    let default_account = mk_account();
+
+    // For mappings where the address is specified, populate the named address mapping
+    let additional_mapping =
+        additional_mapping
+            .into_iter()
+            .chain(accounts.iter().map(|(n, test_account)| {
+                let addr =
+                    NumericalAddress::new(test_account.address.to_inner(), NumberFormat::Hex);
+                (n.clone(), addr)
+            }));
+    // Extend the mappings of all named addresses with values
+    for (name, addr) in additional_mapping {
+        if named_address_mapping.contains_key(&name) || name == "sui" {
+            panic!("Invalid init. The named address '{}' is reserved", name)
+        }
+        named_address_mapping.insert(name, addr);
+    }
+
+    AccountSetup {
+        default_account,
+        named_address_mapping,
+        objects,
+        account_objects,
+        accounts,
+    }
+}
+
+/// Create the executor for a validator with a fullnode
+/// The issue with this executor is we cannot control the checkpoint
+/// and epoch creation process
+async fn init_val_fullnode_executor(
+    mut rng: StdRng,
+    account_names: BTreeSet<String>,
+    additional_mapping: BTreeMap<String, NumericalAddress>,
+    protocol_config: &ProtocolConfig,
+) -> (Box<dyn TransactionalAdapter>, AccountSetup) {
+    let acc_setup = create_accounts_objects(&mut rng, account_names, additional_mapping);
+    let executor = create_val_fullnode_executor(protocol_config, &acc_setup.objects).await;
+    (Box::new(executor), acc_setup)
+}
+
+/// Create an executor using a simulator
+/// This means we can control the checkpoint, epoch creation process and
+/// manually advance clock as needed
+fn init_sim_executor(
+    mut rng: StdRng,
+    account_names: BTreeSet<String>,
+    additional_mapping: BTreeMap<String, NumericalAddress>,
+    protocol_config: &ProtocolConfig,
+) -> (Box<dyn TransactionalAdapter>, AccountSetup) {
+    // Initial list of named addresses with specified values
+    let mut named_address_mapping = NAMED_ADDRESSES.clone();
+    let mut account_objects = BTreeMap::new();
+    let mut accounts = BTreeMap::new();
+    let mut objects = vec![];
+
+    // Closure to create accounts with gas objects of value `GAS_FOR_TESTING`
+    let mut mk_account = || {
+        let (address, key_pair) = get_key_pair_from_rng(&mut rng);
+        let obj = Object::with_id_owner_gas_for_testing(
```

Comment:
Where does this object go?

Reply:
Copy pasta. Will cleanup in followup PR.

Reply:
Here's the pr which cleans this up https://github.com/MystenLabs/sui/pull/14787